### PR TITLE
MHV Landing Page redirect bugfix

### PR DESCRIPTION
**Pull request by radavis**
## Summary

As a veteran who doesn't have access to the new MHV on VA.gov, when I sign in, I want to be directed to myhealth.va.gov/mhv-portal-web/home.

**The replication steps for the bug are:**
1. make sure you are signed out of all va.gov sessions
2. navigate to va.gov/my-health
3. see the mhv landing page app start loading
4. see redirect to myhealth.va.gov/mhv-portal-web/home happen
5. see broken MHV national portal landing page

**The desired behavior/happy path should be:**
1. user is signed out of all va.gov sessions
2. user navigates to va.gov/my-health
3. user is redirected / sees sign in modal
4. users signs in
5. user is redirected to va.gov/my-health
6. if user meets feature toggle criteria, user sees va.gov/my-health
7. if user does not meet criteria, user sees myhealth.va.gov/mhv-portal-web/home

## Related issue(s)

department-of-veterans-affairs/va.gov-team#61496

## Testing done

Add `/my-health` path to URL and press Enter/Return. User sees SSO options. After sign in, authorized users are shown the Landing Page at `/my-health`. Unauthorized users are redirected to MHV National Portal.
## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

**Before**

[61496-mhvlp-bug.webm](https://github.com/department-of-veterans-affairs/vets-website/assets/1299034/97a8459a-0c8d-43cd-bd74-350d68c28ea4)

**After**

[61496-mhvlp-bugfix.webm](https://github.com/department-of-veterans-affairs/vets-website/assets/1299034/46ae33b5-50e3-490f-8ab6-aa0f8ffe697c)

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/landing-page/tests/selectors/isLandingPageEnabledForUser.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/tests/selectors/isLandingPageEnabledForUser.unit.spec.js:1]: Using camelcased feature toggles in this spec file.

File: src/applications/mhv/landing-page/tests/containers/App.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/tests/containers/App.unit.spec.jsx:1]: Using camelcased feature toggles in this spec file.

File: src/applications/mhv/landing-page/containers/App.jsx [hunk 3]
```diff
@@ -1,47 +1,70 @@
...
+    profile.loading;
+
+  const redirecting = signedIn && !loading && !enabled;
+
+  useEffect(
     () => {
-      return isLandingPageEnabledForUser(fullState);
+      const redirect = () => {
+        const redirectUrl = mhvUrl(ssoe, 'home');
+        // console.log({ redirectUrl }); // eslint-disable-line no-console
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/containers/App.jsx:48]: This line was useful for debugging, but can be removed.

File: src/applications/mhv/landing-page/tests/e2e/app-enabled/disabled.cypress.spec.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  beforeEach(() => {
+    cy.intercept('GET', '/data/cms/vamc-ehr.json', vamcEhr).as('vamcEhr');
+    cy.intercept('GET', '/v0/feature_toggles*', featureTogglesDisabled).as(
+      'featureTogglesDisabled',
+    );
+    const mhvAuthRedirectUrl = 'https://pint.eauth.va.gov/mhv-portal-web/eauth';
+    cy.intercept('GET', mhvAuthRedirectUrl, '').as('mhvAuthRedirect');
+  });
+
+  // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/tests/e2e/app-enabled/disabled.cypress.spec.js:16]: `cy.axeCheck()` would be performed on a page outside of va.gov if called in this scenario.

File: src/applications/mhv/landing-page/tests/e2e/mhv-auth-redirect.cypress.spec.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    cy.intercept('GET', '/data/cms/vamc-ehr.json', vamcEhr).as('vamcEhr');
+    cy.intercept('GET', '/v0/feature_toggles*', featureToggles).as(
+      'featureToggles',
+    );
+    const redirectUrl = 'https://pint.eauth.va.gov/mhv-portal-web/eauth';
+    cy.intercept('GET', redirectUrl, '').as('mhvRedirect');
+  });
+
+  describe('unauthenticated user', () => {
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/tests/e2e/mhv-auth-redirect.cypress.spec.js:19]: `cy.axeCheck()` would be performed on a page outside of va.gov if called in this scenario.

File: src/applications/mhv/landing-page/tests/e2e/mhv-auth-redirect.cypress.spec.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    it('renders the landing page', () => {
+      cy.login(user);
+      cy.visit(rootUrl);
+      cy.get('h1').should('include.text', 'My HealtheVet');
+      cy.axeCheck();
+    });
+  });
+
+  describe('Cerner patient', () => {
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/tests/e2e/mhv-auth-redirect.cypress.spec.js:37]: `cy.axeCheck()` would be performed on a page outside of va.gov if called in this scenario.

File: src/applications/mhv/landing-page/tests/e2e/mhv-auth-redirect.cypress.spec.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    it('redirects to MHV National Portal', () => {
+      cy.login(cernerUser);
+      cy.visit(rootUrl);
+      cy.wait('@mhvRedirect');
+    });
+  });
+
+  describe('unauthorized user', () => {
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv/landing-page/tests/e2e/mhv-auth-redirect.cypress.spec.js:46]: `cy.axeCheck()` would be performed on a page outside of va.gov if called in this scenario.

File: src/applications/mhv/landing-page/tests/e2e/restricted-by-log-in-provider/auth.provider.cypress.spec.js [hunk 8]
```diff
@@ -1,32 +1,38 @@
...
+import { appName } from '../../../manifest.json';
+import vamcEhr from '../../fixtures/vamc-ehr.json';
 
 import ApiInitializer from '../utilities/ApiInitializer';
 import LandingPage from '../pages/LandingPage';
 
-describe(manifest.appName, () => {
+describe(appName, () => {
   beforeEach(() => {
+    cy.intercept('GET', '/data/cms/vamc-ehr.json', vamcEhr).as('vamcEhr');
```
Comment (mdewey) [inline src/applications/mhv/landing-page/tests/e2e/restricted-by-log-in-provider/auth.provider.cypress.spec.js:10]: MINOR REFACTOR: I would consider using the ApiInitializer class to keep your intercepts in one place that is reusable 

https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/mhv/landing-page/tests/utilities/ApiInitializer.js

**Review by mdewey — APPROVED** [review]
NITPICK: There are some linting error that could be resolved. 

The small recommendation/nitpicks are not blocking just things to clean up later, this LGTM.

Good work 🚀

File: src/applications/mhv/landing-page/tests/e2e/restricted-by-log-in-provider/auth.provider.cypress.spec.js (hunk 8 above)
Comment (radavis) [reply src/applications/mhv/landing-page/tests/e2e/restricted-by-log-in-provider/auth.provider.cypress.spec.js:10]: Good call. I'll be sure to use and update the ApiInitializer in future PRs.

**Review by powellkerry — APPROVED** [review]

**Pull request merged**